### ISSUE TITLE
Equation 10

### ISSUE THREAD

**Issue by affromero** [zj10/PGA#1]
Hello, 
Nice paper and nice implementation. 
I have a question regarding VPGA code.
These two lines stand for Equation 10, `vrec` and `vkld` for the first and second term, respectively, right?
```python
vrec_loss = tf.losses.mean_squared_error(z_mu_rec / z_mu_norm, z_rec / z_mu_norm)
vkld_loss = -tf.reduce_mean(0.5 * (1 + z_log_sigma_sq - z_mu ** 2 - tf.exp(z_log_sigma_sq)))
```
1. According to Equation 10, why is the first term not negative as it is the second one?
2. Why two different Z inputs ([`z_mu` and `z_mu+z_noise`](https://github.com/zj10/PGA/blob/master/src/train_vpga.py#L113-L114)) must produce the same output by minimizing `vrec_loss` ([z_mu_rec, z_rec](https://github.com/zj10/PGA/blob/master/src/train_vpga.py#L120))
3. Why [`z_mu_norm`](https://github.com/zj10/PGA/blob/master/src/train_vpga.py#L119) (where it comes from?) in `vrec_loss`?

**Comment by zj10** [comment]
Thanks for your interest.

1 & 2. vrec_loss is derived from the Gaussian assumption on \hat{z} (z_rec), and thus is equivalent to minimizing a MSE loss.
3. z_mu_norm is used to normalize vrec_loss to avoid the trivial minimum, where z_mu, z_mu_rec and z_rec are close to zero. We will explain this in later versions of the paper.

**Comment by affromero** [comment]
Thanks for your reply. 
I am not sure if I follow you in 1 and 2:
- z_mu_rec = dec_enc(z_mu)
- z_rec = dec_enc(z_mu+z_noise)

Please correct me if I am wrong: If `z_mu` and `z_mu+z_noise` are two different Gaussian distributions, I do not understand why the output of `dec_enc` should be the same?

**Comment by zj10** [comment]
vrec_loss is just the VAE reconstruction loss moved from the data space to the latent space. The intuition behind is that, vrec_loss tries to expand the volume occupied by each z_mu. It is not necessary for z_mu_rec and z_rec to be the same though. And in practice, they are rarely the same.
